axes/utils.py

```diff
@@ -69,10 +69,12 @@ def get_client_ip(request):
     return getattr(request, client_ip_attribute)
 
 
-def get_client_username(request):
+def get_client_username(request, credentials=None):
     if settings.AXES_USERNAME_CALLABLE:
-        return settings.AXES_USERNAME_CALLABLE(request)
-    return request.POST.get(settings.AXES_USERNAME_FORM_FIELD, None)
+        return settings.AXES_USERNAME_CALLABLE(request, credentials)
+    if credentials is None:
```

Comment:
just a semantics note, but reversing this logic to be `is not None` would make it slightly more clear that the request method is the "fallback"

Reply:
I agree, it would make this more readable as a whole.